### PR TITLE
Fetch performance tests

### PR DESCRIPTION
**Pull request by notkriswagner**
Added set of performance tests to the C API to test fetch. Also added script to populate test data in a specific deployment.

### CODE REVIEW

**Comment by smtakeda** [comment]
started the review. @notkriswagner ensure git rebase.

File: scripts/setup_test_data.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    con.cursor().execute(delete_table_data)
+    con.cursor().execute(put_csv_file)
+    print('CSV file has been uploaded to table stage')
+    con.cursor().execute(copy_into_table)
+    print('Test data has been loaded into table')
+    con.cursor().execute(remove_csv_file)
+else:
+    print('Test data already exists in account, no data population needed')
+
+sys.exit(0)
```
Comment (smtakeda) [inline scripts/setup_test_data.py]: Add a newline to the end.

File: scripts/sf_test_utils.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    if port:
+        params['port'] = port
+    protocol = os.getenv("SNOWFLAKE_TEST_PROTOCOL")
+    if protocol:
+        params['protocol'] = protocol
+    warehouse = os.getenv("SNOWFLAKE_TEST_WAREHOUSE")
+    if warehouse:
+        params['warehouse'] = warehouse
+
+    return params
```
Comment (smtakeda) [inline scripts/sf_test_utils.py]: Add a newline to the end.

This often cause unnecessary diff when somebody else edits the file.

**Comment by smtakeda** [comment]
Aside from PR, clean up merged branches:
https://github.com/snowflakedb/libsnowflakeclient/branches

File: tests/test_perf_string_reads_and_writes.c [hunk 3]
```diff
@@ -0,0 +1,238 @@
+/*
```
Comment (smtakeda) [inline tests/test_perf_string_reads_and_writes.c]: How do we run the perf tests? Do they run on every TravisCI or we would have to manually run?
Also please share the perf test results in the team.

File: tests/CMakeLists.txt [hunk 4]
```diff
@@ -125,3 +129,23 @@ FOREACH(T ${TESTS_CXX})
...
+        add_executable(${T} ${SOURCE_UTILS} ${T}.c)
+        target_include_directories(
+                ${T} PUBLIC
+                ../deps-build/${PLATFORM}/cmocka/include
+        )
+        target_link_libraries(${T} ${TESTLIB_OPTS_C})
+        add_test(${T} ${T})
+        # Removed to suppress false alarm: "--run-libc-freeres=no"
+        add_test(valgrind_${T}
+                valgrind
```
Comment (howryu) [inline tests/CMakeLists.txt]: In general, running test with valgrind will hurt performance. If you are adding performance tests, why are you running with valgrind?

File: tests/utils/test_setup.c [hunk 5]
```diff
@@ -58,3 +58,38 @@ void dump_error(SF_ERROR_STRUCT *error) {
...
+      query,
+      0);
+    if (status != SF_STATUS_SUCCESS) {
+        dump_error(&(sfstmt->error));
+    }
+    assert_int_equal(status, SF_STATUS_SUCCESS);
+}
+
+void process_results(struct timespec begin, struct timespec end, int num_iterations, const char *label) {
+    FILE *results_file = fopen("/tmp/test_results.csv", "a+");
```
Comment (howryu) [inline tests/utils/test_setup.c]: I don't think writing test results to tmp directory is a good idea. Probably consider some other directory, like some directory in libsnowflakeclient workspace.

File: tests/test_perf_string_reads_and_writes.c (hunk 3 above)
Comment (notkriswagner) [reply tests/test_perf_string_reads_and_writes.c]: They run on every TravisCI, but the results will not be recorded.

File: tests/CMakeLists.txt (hunk 4 above)
Comment (notkriswagner) [reply tests/CMakeLists.txt]: Just to make sure that the performance tests don't have any memory leaks. I don't collect any test results with valgrind.

**Comment by notkriswagner** [comment]
Updated code according to suggestions. If everything looks good to the reviewers, I'll do a rebase and then merge.
